L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -0,0 +1,1095 @@
+  packed_value |= ((waferV & HGCalModuleDetId::kHGCalModuleVMask) << HGCalModuleDetId::kHGCalModuleVOffset);
+  packed_value |= ((subdet & HGCalModuleDetId::kHGCalTriggerSubdetMask) << HGCalModuleDetId::kHGCalTriggerSubdetOffset);
+  packed_value |= ((layer & HGCalModuleDetId::kHGCalLayerMask) << HGCalModuleDetId::kHGCalLayerOffset);
+  return packed_value;
+}
+
+void HGCalTriggerGeometryV9Imp3::unpackLayerSubdetWaferId(
+    unsigned wafer, unsigned& layer, int& subdet, int& waferU, int& waferV) const {
+  waferU = (wafer >> HGCalModuleDetId::kHGCalModuleUOffset) & HGCalModuleDetId::kHGCalModuleUMask;
+  waferV = (wafer >> HGCalModuleDetId::kHGCalModuleVOffset) & HGCalModuleDetId::kHGCalModuleVMask;
+  subdet = (wafer >> HGCalModuleDetId::kHGCalTriggerSubdetOffset) & HGCalModuleDetId::kHGCalTriggerSubdetMask;
+  layer = (wafer >> HGCalModuleDetId::kHGCalLayerOffset) & HGCalModuleDetId::kHGCalLayerMask;
+}
+
+void HGCalTriggerGeometryV9Imp3::etaphiMappingFromSector0(int& ieta, int& iphi, unsigned sector) const {
+  if (sector == 0) {
+    return;
+  }
+  iphi = iphi + (sector * hSc_num_panels_per_sector_);
+}
+
+HGCalGeomRotation::WaferCentring HGCalTriggerGeometryV9Imp3::getWaferCentring(unsigned layer, int subdet) const {
+  if (subdet == HGCalTriggerSubdetector::HGCalEETrigger) {  // CE-E
+    return HGCalGeomRotation::WaferCentring::WaferCentred;
+  } else if (subdet == HGCalTriggerSubdetector::HGCalHSiTrigger) {
+    if ((layer % 2) == 1) {  // CE-H Odd
+      return HGCalGeomRotation::WaferCentring::CornerCentredY;
+    } else {  // CE-H Even
+      return HGCalGeomRotation::WaferCentring::CornerCentredMercedes;
+    }
+  } else if (subdet == HGCalTriggerSubdetector::HFNoseTrigger) {  //HFNose
+    return HGCalGeomRotation::WaferCentring::WaferCentred;
+  } else {
+    edm::LogError("HGCalTriggerGeometryV9Imp3")
+        << "HGCalTriggerGeometryV9Imp3: trigger sub-detector expected to be silicon";
+    return HGCalGeomRotation::WaferCentring::WaferCentred;
+  }
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::etaphiMappingToSector0(int& ieta, int& iphi) const {
+  unsigned sector = 0;
+
+  if (unsigned(std::abs(iphi)) > 2 * hSc_num_panels_per_sector_)
+    sector = 2;
+  else if (unsigned(std::abs(iphi)) > hSc_num_panels_per_sector_)
+    sector = 1;
+  else
+    sector = 0;
+
+  iphi = iphi - (sector * hSc_num_panels_per_sector_);
+
+  return sector;
+}
+
+bool HGCalTriggerGeometryV9Imp3::validTriggerCell(const unsigned trigger_cell_id) const {
+  return validTriggerCellFromCells(trigger_cell_id);
+}
+
+bool HGCalTriggerGeometryV9Imp3::disconnectedModule(const unsigned module_id) const {
+  bool disconnected = false;
+  HGCalModuleDetId id(module_id);
+  if (module_to_stage1_.find(packLayerSubdetWaferId(id.layer(), id.triggerSubdetId(), id.moduleU(), id.moduleV())) ==
+      module_to_stage1_.end()) {
+    disconnected = true;
+    return disconnected;
```

Comment:
Not needed. The `return` at the end is enough

Reply:
thanks, removed